fs.go

```diff
@@ -13,6 +13,37 @@ var (
 	ErrCrossedBoundary = errors.New("chroot boundary crossed")
 )
 
+// Capability holds the supported features of a billy filesystem. This does
+// not mean that the capability has to be supported by the underlying storage.
+// For example, a billy filesystem may support CapWrite but the storage be
+// mounted in read only mode.
+type Capability uint64
+
+const (
+	// CapWrite means that the fs is writable.
+	CapWrite Capability = 1 << iota
```

Comment:
I would change the names to `WriteCapability` and so on.

Reply:
Done